apps/aecore/src/aec_blocks.erl

```diff
@@ -162,3 +163,40 @@ deserialize_from_map(#{<<"height">> := Height,
 hash_internal_representation(B = #block{}) ->
     aec_headers:hash_header(to_header(B)).
 
+-spec validate(block()) -> ok | {error, term()}.
+validate(Block) ->
+    Validators = [fun validate_coinbase_txs_count/1,
+                  fun validate_txs_hash/1],
+    aeu_validation:run(Validators, Block).
+
+-spec validate_coinbase_txs_count(block()) -> ok | {error, multiple_coinbase_txs}.
+validate_coinbase_txs_count(#block{txs = Txs}) ->
+    CoinbaseTxsCount = lists:filter(
```

Comment:
Why not do a fold over integers if only is the count you want?

Reply:
Good point.